lib/smart_answer/calculators/benefit_cap_calculator_configuration.rb

```diff
@@ -38,9 +38,19 @@ def descriptions(version)
     end
 
     def all_questions
-      dataset.keys.inject({}){|versions, version| versions.merge(questions(version))}
+      dataset.keys.inject({}) { |versions, version| versions.merge(questions(version)) }
     end
 
+    def is_london?(postcode)
+      area(postcode).any? { |result| result[:slug] == "london" }
+    end
+
+    def area(postcode)
+      response = Services.imminence_api.areas_for_postcode(postcode)
+      (response&.results || {})
+    end
```

Comment:
@leenagupte:

I'm not quite sure I understand what you mean. I just tried the following in the rails console and it looks as if `#areas_for_postcode` returns an instance of `GdsApi::Response` which *does* have a `#code` method, even though this doesn't show up in the string generated by `#inspect`.

```
response = Services.imminence_api.areas_for_postcode('EC1A 1AA')
# => #<GdsApi::Response:0x007fa4d40a75f0 @http_response="{\"_response_info\":{\"status\":\"ok\",\"links\":[]},\"total\":14,\"start_index\":1,\"page_size\":14,\"current_page\":1,\"pages\":1,\"results\":[{\"slug\":\"house-of-commons\",\"name\":\"House of Commons\",\"country_name\":\"-\",\"type\":\"WMP\"},{\"slug\":\"islington-022\",\"name\":\"Islington 022\",\"country_name\":\"England\",\"type\":\"OMG\"},{\"slug\":\"london-assembly\",\"name\":\"London Assembly\",\"country_name\":\"England\",\"type\":\"LAE\"},{\"slug\":\"islington-022f\",\"name\":\"Islington 022F\",\"country_name\":\"England\",\"type\":\"OLF\"},{\"slug\":\"london-assembly\",\"name\":\"London Assembly\",\"country_name\":\"England\",\"type\":\"LAS\"},{\"slug\":\"greater-london-authority\",\"name\":\"Greater London Authority\",\"country_name\":\"England\",\"type\":\"GLA\"},{\"slug\":\"islington-borough-council\",\"name\":\"Islington Borough Council\",\"country_name\":\"England\",\"type\":\"LBO\"},{\"slug\":\"islington-022f\",\"name\":\"Islington 022F\",\"country_name\":\"England\",\"type\":\"OLG\"},{\"slug\":\"clerkenwell\",\"name\":\"Clerkenwell\",\"country_name\":\"England\",\"type\":\"LBW\"},{\"slug\":\"european-parliament\",\"name\":\"European Parliament\",\"country_name\":\"-\",\"type\":\"EUP\"},{\"slug\":\"islington-022\",\"name\":\"Islington 022\",\"country_name\":\"England\",\"type\":\"OMF\"},{\"slug\":\"north-east\",\"name\":\"North East\",\"country_name\":\"England\",\"type\":\"LAC\"},{\"slug\":\"islington-south-and-finsbury\",\"name\":\"Islington South and Finsbury\",\"country_name\":\"England\",\"type\":\"WMC\"},{\"slug\":\"london\",\"name\":\"London\",\"country_name\":\"England\",\"type\":\"EUR\"}]}">
response.code
# => 200
```

However, I'm not very familiar with `GdsApi::Imminence` and it's quite possible that the implementation in `LandlordImmigrationCheckCalculator` might be incorrect as I think you're suggesting.

Either way, I don't understand why the implementation in `BenefitCapCalculatorConfiguration` and `LandlordImmigrationCheckCalculator` should end up being very different from each other. If the implementation in the latter is wrong, then I think the solution is to fix it and bring it into line with that in `LandlordImmigrationCheckCalculator`.